support/main/default/classes/TriggerControlTest.cls

```diff
@@ -1,137 +1,243 @@
+    @isTest
+    private static void invalidHandlerTest() {
+        Map<String, TriggerConfigService.ObjectConfig> mdt = buildMetadata();
+        TriggerConfigService.ObjectConfig objConfig = mdt.get('contact');
+        objConfig.disabledHandlers.clear();
+        objConfig.handlerOrder.clear();
+        objConfig.handlers.clear();
+        TriggerConfigService.HandlerConfig handlerConfig = generateHandlerTestData('badNameClass', true);
+        addChildHandlerToObj(objConfig, handlerConfig);
+        mdt.put('contact', objConfig);
+        TriggerConfigService.configMetadata = mdt;
+        // invalid class name should not run - currently just debugs
+        Boolean acctHandler1Fired = true;
+        Boolean acctHandler2Fired = true;
+        Boolean contactHandler1Fired = false;
+        runTest(acctHandler1Fired, acctHandler2Fired, contactHandler1Fired);
+    }
+
+    @isTest
+    private static void additionalCoverage() {
+        Map<String, TriggerConfigService.ObjectConfig> forceGet = TriggerConfigService.configMetadata;
+        Trigger_Object__mdt objMdt = new Trigger_Object__mdt(Object__c = 'testerObj', Enabled__c = true);
+        TriggerConfigService.ObjectConfig objConstructorTest = new TriggerConfigService.ObjectConfig(objMdt);
+        Trigger_Handler__mdt handlerMdt = new Trigger_Handler__mdt(Handler_Class__c = 'testerHandler');
+        TriggerConfigService.HandlerConfig handlerConstructorTest = new TriggerConfigService.HandlerConfig(handlerMdt);
+    }
+
+    private static void runTest(Boolean acctHandler1Fired, Boolean acctHandler2Fired, Boolean contactHandler1Fired) {
+        system.debug(TriggerConfigService.configMetadata);
+        Id acctId = createAccount();
+        checkAccount(acctId, acctHandler1Fired, acctHandler2Fired);
+        Id contactId = createContact();
+        checkContact(contactId, contactHandler1Fired);
+    }
+
+    // NB: current version replaces all trigger controls with each call, which
+    // is plenty for the current test methods
+    private static TriggerConfigService.ObjectConfig generateObjTestData(String objName, Boolean enabled) {
+        TriggerConfigService.ObjectConfig trigObj = new TriggerConfigService.ObjectConfig();
+        trigObj.objName = objName;
+        trigObj.enabled = enabled;
+        return trigObj;
+    }
+
+    // Add additional params for each trigger evt?
+    private static TriggerConfigService.HandlerConfig generateHandlerTestData(String className, Boolean enabled) {
+        TriggerConfigService.HandlerConfig handler = new TriggerConfigService.HandlerConfig();
+        handler.className = className;
+        handler.enabled = enabled;
+        handler.autoBind = true;
+        handler.order = 0;
+        handler.beforeInsert = true;
+        handler.afterInsert = false;
+        handler.beforeDelete = true;
+        handler.afterDelete = false;
+        handler.afterUndelete = false;
+        handler.beforeUpdate = true;
+        handler.afterUpdate = false;
```

Comment:
seems unexpected that only befores are enabled, would think they'd all be disabled and caller would be responsible for setting this

Reply:
Yep, I like that. Leave the constructor empty, everything is false, calling method is responsible for identifying which records to run.